metacat-main/src/main/java/com/netflix/metacat/main/manager/ConnectorManager.java

```diff
@@ -41,26 +42,39 @@
 import com.netflix.metacat.common.server.connectors.exception.CatalogNotFoundException;
 import com.netflix.metacat.common.server.properties.Config;
 import com.netflix.metacat.common.server.spi.MetacatCatalogConfig;
+import lombok.AllArgsConstructor;
+import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 
 import javax.annotation.Nonnull;
 import javax.annotation.PreDestroy;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
 
 /**
  * Connector manager.
  */
 @Slf4j
 public class ConnectorManager {
+    private static final String EMPTY_STRING = "";
     // Map of connector plugins registered.
     private final ConcurrentMap<String, ConnectorPlugin> plugins = new ConcurrentHashMap<>();
-    // Map of connector factories registered.
-    private final ConcurrentMap<String, ConnectorFactory> connectorFactories = new ConcurrentHashMap<>();
-    // Map of catalogs registered.
-    private final ConcurrentHashMap<String, MetacatCatalogConfig> catalogs = new ConcurrentHashMap<>();
+    /**
+     * Table of catalog name, database name to catalog data stores. Usually there is a one-to-one mapping between the
+     * catalog name and the data store. In this case the table will have the databse name as null. If there are multiple
+     * catalogs addressed by the same <code>catalog.name</code> pointing to shards of a data store, then there will be
+     * multiple entries in this table. An entry will be identified using the catalog name and the database name.
+     */
+    private final Table<String, String, CatalogHolder> catalogs = HashBasedTable.create();
+    private final Set<MetacatCatalogConfig> catalogConfigs = Sets.newHashSet();
+    private final Set<ConnectorDatabaseService> databaseServices = Sets.newHashSet();
+    private final Set<ConnectorTableService> tableServices = Sets.newHashSet();
+    private final Set<ConnectorPartitionService> partitionServices = Sets.newHashSet();
     private final AtomicBoolean stopped = new AtomicBoolean();
     private final Config config;
```

Comment:
I feel we should factor out the catalogRelated operation into a new class, separating from the database/table/part services related informations. The catalog class has 
`private final Table<String, String, CatalogHolder> catalogs = HashBasedTable.create();`
and catalog related information.